### PR TITLE
maxTextLength API

### PR DESCRIPTION
**Pull request by lxgreen**
### :rocket: New Feature
- `editor`
  - maxTextLength prop added

### CODE REVIEW

File: docs/docs/ricos/ricos-api.mdx [hunk 1]
```diff
@@ -200,6 +200,10 @@ By default, its predicate is `relValue !== 'nofollow'`
 Determines the placeholder to use for the link panel.
 By default, its value is `"Enter a URL here"` translated by localization
 
+### `maxTextLength`
+
+Limits the editor text length to a given value. When the max text length is reached, further text input is ignored. Also, text paste action is canceled if the resulting text would exceed the max length.
```
Comment (shaulgo) [inline docs/docs/ricos/ricos-api.mdx:205]: This is exactly like html. More info only confusing?
```suggestion
Limits the editor text length.
```

File: packages/editor/web/src/RichContentEditor/handlePastedText.js [hunk 2]
```diff
@@ -146,8 +175,23 @@ export const convertParsedEditorStateObjectToRawData = rawContent => {
...
+  editorState,
+  pasteWithoutAtomic,
+  customHeadings,
+  maxTextLength,
+  getSelectedText,
+}) => {
   return isCopyFromEditor(html) && !pasteWithoutAtomic
-    ? handlePastedTextFromEditor(html, editorState)
-    : handlePastedTextFromOutsideEditor(text, normalizeHTML(html), editorState, customHeadings);
+    ? handlePastedTextFromEditor(html, editorState, maxTextLength, getSelectedText)
```
Comment (shaulgo) [inline packages/editor/web/src/RichContentEditor/handlePastedText.js:188]: maxLength should probably be only here, instead of both functions.

**Review by shaulgo — CHANGES_REQUESTED** [review]

File: docs/docs/ricos/ricos-api.mdx (hunk 1 above)
Comment (lxgreen) [reply docs/docs/ricos/ricos-api.mdx:205]: 👍

File: packages/editor/web/src/RichContentEditor/RichContentEditor.tsx [hunk 3]
```diff
@@ -717,6 +730,20 @@ class RichContentEditor extends Component<RichContentEditorProps, State> {
       // in general, disables any input click on atomic blocks
       handled = 'handled';
     }
+
+    // input is ignored if length > maxTextLength (if maxTextLength is set)
+    if (this.props.maxTextLength && this.props.maxTextLength > 0) {
+      const contentLength = this.state.editorState.getCurrentContent().getPlainText('').length;
+      const selectedTextLength = this.getSelectedText(this.state.editorState).length;
```
[the remark this answers is not in the document]
Comment (lxgreen) [reply packages/editor/web/src/RichContentEditor/RichContentEditor.tsx:737]: hmm only if `import()` fails in `preloadLibs`

File: packages/editor/web/src/RichContentEditor/handlePastedText.js (hunk 2 above)
Comment (lxgreen) [reply packages/editor/web/src/RichContentEditor/handlePastedText.js:188]: why? what if it's copied from outside (e.g. Word) -- i still want to limit the length. AFAIK, `isCopyFromEditor` filters these cases, right?

**Pull request merged**